x/concentrated-liquidity/architecture.md

```diff
@@ -77,22 +77,102 @@ $$\Delta x = \Delta \frac {1}{\sqrt P}  L$$
 
-$$\sqrt{p(-1)} = 1.0001^{-1/2} \approx 0.99995$$
+Since we know what range a pair will generally trade in, how do we go about providing more granularity at that range and provide a more optimal price range between ticks instead of the "one-size-fits-all" approach explained above?
 
-$$\sqrt{p(0)} = 1.0001^{0/2} = 1$$
+### Geometric Tick Spacing with Additive Ranges
 
-$$\sqrt{p(1)} = \sqrt{1.0001} = 1.0001^{1/2} \approx 1.00005$$
+In Osmosis' implementation of concentrated liquidity, we will instead make use of geometric tick spacing with additive ranges.
 
-TODO: tick range bounds
+We start by defining an exponent for the precision factor of 10 at a spot price of one - ($exponentAtPriceOne$).
+
+For instance, if $exponentAtPriceOne = -4$ , then each tick starting at 1 and ending at the first factor of 10 will represents a spot price increase of 0.0001. At this precision factor:
+* $tick_0 = 1$ (tick 0 is always equal to 1 regardless of precision factor)
+* $tick_1 = 1.0001$
+* $tick_2 = 1.0002$
+* $tick_3 = 1.0003$
+
+This continues on until we reach a spot price of 10. At this point, since we have increased by a factor of 10, our $exponentAtCurrentTick$ increases from -4 to -3, and the ticks will increase as follows:
+* $tick_{89999} =  9.9999$
+* $tick_{90000} = 10.000$
+* $tick_{90001} = 10.001$
+* $tick_{90002} = 10.002$
+
+For spot prices less than a dollar, the precision factor decreases at every factor of 10. For example, with a $exponentAtPriceOne$ of -4:
+* $tick_{-1} = 0.9999$
+* $tick_{-2} = 0.9998$
+* $tick_{-5001} = 0.4999$
+* $tick_{-5002} = 0.4998$
+
+With a $exponentAtPriceOne$ of -6:
+* $tick_{-1} = 0.999999$
+* $tick_{-2} = 0.999998$
+* $tick_{-5001} = 0.994999$
+* $tick_{-5002} = 0.994998$
+
+### Formulas
+
+After we define $exponentAtPriceOne$ (this is chosen by the user based on what precision they desire the asset pair to trade at), we can then calculate how many ticks must be crossed in order for k to be incremented ($geometricExponentIncrementDistanceInTicks$).
+
+$$ geometricExponentIncrementDistanceInTicks = 9 * 10^{(-exponentAtPriceOne)} $$
+
+Since we define k at price one and utilize this as the increment starting point instead of price zero, we must multiply the result by 9 as shown above. In other words, starting at 1, it takes 9 ticks to get to the first power of 10. Then, starting at 10, it takes 9*10 ticks to get to the next power of 10, etc.
+
+Now that we know how many ticks must be crossed in order for our k to be incremented, we can then figure out what our change in k will be based on what tick we are trading at:
+
+$$ geometricExponentDelta = ⌊ tick / geometricExponentIncrementDistanceInTicks ⌋ $$
```

Comment:
Lets say `geometricExponentIncrementDistanceInTicks = 9` for now.

* `tick 9 -> 1`, 
* `tick 5 -> 0`, 
* `tick 0 -> 0`, 
* `tick -1 -> -1`, 
* `tick -9 -> -1`,
* `tick -10 -> -2`

under this algo, which looks right to me. However in the code it does:
```
	if tickIndex.IsNegative() {
		geometricExponentIncrementDistanceInTicks = geometricExponentIncrementDistanceInTicks.Quo(sdkTenDec)
	}
```

How com?

Reply:
From the previous discussion we had, this is a quote from the hackmd you wrote:

"Say k=-2. We go 1, then 1.01, then 1.02. But on the downside, should we increment k or not? e.g. is it 1, .99, .98 OR 1, .999, .998"

The code you posted implements the latter

Reply:
@ValarDragon I remember why I did this now. Without the above line, we can only get to .000000000000000010 precision instead of .000000000000000001. Unsure if that is necessary

Reply:
Following up on this, ended up implementing the stair step precision as discussed